streamz/core.py

```diff
@@ -181,6 +181,24 @@ def _inform_asynchronous(self, asynchronous):
                 if downstream:
                     downstream._inform_asynchronous(asynchronous)
 
+    def _add_upstream(self, upstream):
```

Comment:
Although this is internal API, I think one-line docstrings would be useful for developers.
Also, I suspect that the old way (self.upstream.downstreams.add/remove) is used in many places - do you think they ought to be changed to use this formalism now?

Reply:
I think the classes which use the old way should be migrated, otherwise they might run into issues with zip and such. I'll look to see if that pattern is used elsewhere.

Reply:
The new `slice` method has it, sorry: https://github.com/python-streamz/streamz/blob/master/streamz/core.py#L795

Reply:
Would appreciate if you merged from master and made sure that `slice` works

Reply:
ping on this one

Reply:
sorry about that, forgot it was there!

Reply:
ping...

Reply:
I think this is taken care of now? https://github.com/python-streamz/streamz/pull/248/files#diff-d4e512993f47710ea9e7155d66011a24R813

Reply:
Oh I am sorry - I don't know why I didn't see your commits come in.